projects/optic/src/commands/capture/capture.ts

```diff
@@ -581,6 +581,13 @@ async function startApp(
   app.stderr.on('data', (data) => {
     logger.error(data.toString());
   });
+
+  app.on('exit', (code) => {
+    if (code! > 0) {
+      throw `Unexpected exit from the server with exit code: ${code}`;
+    }
+  });
```

Comment:
So I'm not sure this will work as we expect - it's throwing inside of the fn here which may error after the server starts and will throw in a separate function stack

Instead I think this should be set _outside_ of this function and we should be tracking whether we expect the code to fail (i.e. with a variable `expectAppRunning`)

Reply:
Ok so this still exits, but it results in an uncaught exception - I'm not sure what the best way to handle this is right now - will need to try some things to fix this - we can probably merge it and add a note here to handle this

Also we should throw ```new Error(`Unexpected exit from the server with exit code: ${code}`)``` instead

Reply:
commented it for now 👍